OpenRA.Mods.Common/Traits/AttackMove.cs

```diff
@@ -97,14 +97,7 @@ string IOrderVoice.VoicePhraseForOrder(Actor self, Order order)
 		void Activate(Actor self, bool assaultMove)
 		{
 			assaultMoving = assaultMove;
-			self.QueueActivity(new AttackMoveActivity(self, move.MoveTo(TargetLocation.Value, 1)));
-		}
-
-		void INotifyIdle.TickIdle(Actor self)
-		{
-			// This might cause the actor to be stuck if the target location is unreachable
-			if (TargetLocation.HasValue && self.Location != TargetLocation.Value)
-				Activate(self, assaultMoving);
+			self.QueueActivity(new AttackMoveActivity(self, TargetLocation.Value, 1));
 		}
 
 		public void ResolveOrder(Actor self, Order order)
```

Comment:
I think you can get rid of `TargetLocation` now. Unless I missed something it's only used once anymore, and that's in the same method that it's being set.

Also `Activate()` can be removed assimilated into `ResolveOrder()`.

Reply:
Okay. I got rid of them now.